pkg/spoke/submarineragent/config_controller.go

```diff
@@ -249,9 +250,10 @@ func (c *submarinerConfigController) cleanupClusterEnvironment(ctx context.Conte
 		}
 
 		return errors.WithMessagef(err, "Failed to clean up the submariner cluster environment")
-	}
-
-	if config.Status.ManagedClusterInfo.Platform != "AWS" {
+	case "AWS":
+		// Nothing to do
```

Comment:
You can add a comment that cloud-prepare for AWS is handled on Hub, so nothing to be done here (aka Spoke)